test/e2e_node/docker_test.go

```diff
@@ -70,4 +77,174 @@ var _ = framework.KubeDescribe("Docker features [Feature:Docker]", func() {
 			}
 		})
 	})
+
+	Context("when live-restore is enabled", func() {
+		It("containers should not be disrupted when the daemon is down [ygg]", func() {
+			const (
+				startTimeFilename   = "start-timestamp"
+				currentTimeFilename = "current-timestamp"
+			)
+			isSupported, err := isDockerLiveRestoreSupported()
+			framework.ExpectNoError(err)
+			if !isSupported {
+				framework.Skipf("Docker live-restore is not supported.")
+			}
+
+			By("Check whether live-restore is enabled.")
+			isEnabled, err := isDockerLiveRestoreEnabled()
+			framework.ExpectNoError(err)
+			framework.Logf("Docker live-restore enabled: %t", isEnabled)
+			if !isEnabled {
+				// Enables Docker live-restore if it's supported but disabled.
+				By("Enable Docker live-restore.")
+				framework.ExpectNoError(setDockerLiveRestore(true))
+
+				b, err := isDockerLiveRestoreEnabled()
+				framework.ExpectNoError(err)
+				if !b {
+					framework.Logf("Failed to enable Docker live-restore: %v", err)
+				}
+
+				defer func() {
+					By("Disable Docker live-restore.")
+					time.Sleep(2 * time.Second)
+					framework.ExpectNoError(setDockerLiveRestore(false))
+				}()
+			}
+
+			// Creates a temporary directory that will be mounted into the
+			// container, serving as the communication channel between the host
+			// and the container.
+			By("Create temporary directory for mount.")
+			tempDir, err := ioutil.TempDir("", "")
+			framework.ExpectNoError(err)
+			defer func() {
+				By("Remove temporary directory.")
+				defer os.RemoveAll(tempDir)
+			}()
+
+			// Creates a container that
+			//   - writes the current timestamp every second to the
+			//     current-timestamp file, and
+			//   - writes the start timestamp to the start-timestamp file
+			// in the directory that's mounted from the host machine.
+			//
+			// We will be able to tell
+			//   - whether the container is running by checking if the
+			//     current-timestamp increases, and
+			//   - whether the container has been restarted by checking if the
+			//     start-timestamp has changed.
+			cmd := "" +
+				// Writes the start time on start.
+				"date +%s > /test-dir/" + startTimeFilename + "; " +
+				// Writes the current timestamp every second.
+				"while true; do " +
+				"    date +%s > /test-dir/" + currentTimeFilename + "; " +
+				"    sleep 1; " +
+				"done"
```

Comment:
I haven't tried using logs and the documentation doesn't say it will be disrupted. I think what we do here is similar as using logs? To access the logs while the docker is down, we need to read from `/var/lib/docker/containers/`?